### ISSUE TITLE
What are the necessary files in the git repo to run the stack assuming the required images are built?

### ISSUE THREAD

**Issue by timothyjmtan** [openfaas/faas#1089]
Hello,

I am excited to try OpenFaaS at my workplace.

I understand the command to kick-start the entire process is via running the `deploy_stack.sh`. However  as my office environment is a highly restrictive one,, it would be more convenient instead to only download the essential files required to run the stack instead of cloning the entire github library as suggested in the docs:
https://github.com/openfaas/workshop/blob/master/lab1.md#deploy-openfaas

Assuming that I already have the necessary images built, am I able to deploy the stack with only the 'docker-compose.yml' config file, or are the other files in the repository required?

In addition, `docker stack deploy` requires docker version 1.25+. Our corporate environment only has v1.13.1 as the latest one. Am I able to use a plain vanilla `docker-compose` command, or could you recommend any alternatives I could use?

cf: https://docs.docker.com/engine/reference/commandline/stack_deploy/

> API 1.25+ : The client and daemon API must both be at least 1.25 to use this command. Use the docker version command on the client to check your client and daemon API versions.

**Comment by burtonr** [comment]
There are a couple of additional files you'll need to run the stack. If you look at the bottom of the `docker-compose.yml` file, you'll see these referenced:

```yaml
configs:
     prometheus_config:
         file: ./prometheus/prometheus.yml
     prometheus_rules:
         file: ./prometheus/alert.rules.yml
     alertmanager_config:
         file: ./prometheus/alertmanager.yml
```
In other words, you will need 3 additional files found in the `prometheus` directory. You'll need to either update the `docker-compose.yml` file with the correct relative path, or place the files in a `prometheus` directory to match the config file (which I recommend)
* [prometheus.yml](https://github.com/openfaas/faas/blob/master/prometheus/prometheus.yml)
* [alert.rules.yml](https://github.com/openfaas/faas/blob/master/prometheus/alert.rules.yml)
* [alertmanager.yml](https://github.com/openfaas/faas/blob/master/prometheus/alertmanager.yml)

> There are more if you're looking to run on Kubernetes

As for the Docker version, have you tried to run the command with Docker 1.13? `docker stack deploy` was introduced with that version. Noted in the [release blog post](https://blog.docker.com/2017/01/whats-new-in-docker-1-13/)

The warning you're seeing is for the **API version** which are generally much higher versions than the Docker daemon. In the issue template you had to delete to post your message, there's a section to show the output of `docker version`. That output will show the Docker version, API version, and other info. This is mine:

```
$ docker version
Client:
 Version:           18.09.1
 API version:       1.39
 Go version:        go1.10.6
 Git commit:        4c52b90
 Built:             Wed Jan  9 19:35:31 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.1
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.6
  Git commit:       4c52b90
  Built:            Wed Jan  9 19:02:44 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Comment by burtonr** [comment]
Derek add label: question

**Comment by timothyjmtan** [comment]
Thanks @burtonr , that's my mistake and I just realised that I'm currently at API version 1.26! The command should work.

Thanks for the heads-up on the 3 additional .yml files, I've noted that I will have to manage those too.

Just curious, would I be able to exclude Prometheus from the stack if I don't intend to use it?

**Comment by burtonr** [comment]
Great @timothyjmtan! No, you cannot exclude Prometheus. The `docker-compose.yml` contains the minimal configuration to run OpenFaaS on Swarm.

Prometheus' Alert Manager component is what drives the auto-scaling, and thus integral to the operation of OpenFaaS

**Comment by timothyjmtan** [comment]
Awesome, thanks @burtonr  for the extremely speedy and helpful replies! That's all I had to know -- will close the ticket now. 

Thanks again! 👍

**Comment by alexellis** [comment]
I'm unclear why you wouldn't clone the GitHub repository?

I can't really understand this statement?

> my office environment is a highly restrictive one

I've worked behind a very restrictive proxy, but none of them blocked HTTPS to GitHub?

It's relatively small, and if you don't want to do that you can also download the master.zip file from GitHub's UI directly?

Alex

**Comment by alexellis** [comment]
@timothyjmtan why did you delete the whole issue template? Can you give us the data requested to help advise you?

```
<!--- Provide a general summary of the issue in the Title above -->

## Expected Behaviour
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->

## Current Behaviour
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->

## Possible Solution
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

## Steps to Reproduce (for bugs)
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
1.
2.
3.
4.

## Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* FaaS-CLI version ( Full output from: `faas-cli version` ):

* Docker version `docker version` (e.g. Docker 17.0.05 ):

* Are you using Docker Swarm or Kubernetes (FaaS-netes)?

* Operating System and version (e.g. Linux, Windows, MacOS):

* Link to your project or a code example to reproduce issue:

* Please also follow the [troubleshooting guide](https://github.com/openfaas/faas/blob/master/guide/troubleshooting.md) and paste in any other diagnostic information you have:


```

Alex

**Comment by timothyjmtan** [comment]
> I'm unclear why you wouldn't clone the GitHub repository?
> 
> I can't really understand this statement?
> 
> > my office environment is a highly restrictive one
> 
> I've worked behind a very restrictive proxy, but none of them blocked HTTPS to GitHub?
> 
> It's relatively small, and if you don't want to do that you can also download the master.zip file from GitHub's UI directly?
> 
> Alex

Hi Alex, 

Access to github is entirely blocked by our office firewall. In place of that, we have our own internal corporate github and docker repository.

Best,
Timothy

**Comment by timothyjmtan** [comment]
> @timothyjmtan why did you delete the whole issue template? Can you give us the data requested to help advise you?
> 
> ```
> 
> 
> ## Expected Behaviour
> 
> 
> 
> ## Current Behaviour
> 
> 
> 
> ## Possible Solution
> 
> 
> 
> ## Steps to Reproduce (for bugs)
> 
> 
> 1.
> 2.
> 3.
> 4.
> 
> ## Context
> 
> 
> 
> ## Your Environment
> 
> * FaaS-CLI version ( Full output from: `faas-cli version` ):
> 
> * Docker version `docker version` (e.g. Docker 17.0.05 ):
> 
> * Are you using Docker Swarm or Kubernetes (FaaS-netes)?
> 
> * Operating System and version (e.g. Linux, Windows, MacOS):
> 
> * Link to your project or a code example to reproduce issue:
> 
> * Please also follow the [troubleshooting guide](https://github.com/openfaas/faas/blob/master/guide/troubleshooting.md) and paste in any other diagnostic information you have:
> ```
> Alex

Hi Alex,

Sorry about that, my apologies. There were some OSS repositories I used to frequent which have a default template to fill in but wasn't entire obligatory. I'll note the compulsory nature of this one.

In terms of advice being sought, @burtonr has answered all of my outstanding queries. I'll open a new ticket in future if I have others.

Cheers.

Timothy

**Comment by alexellis** [comment]
I still have some outstanding queries. please can you provide the technical information that is asked for in the last section of the template?
